src/js/components/DataTable/Body.js

```diff
@@ -43,177 +43,190 @@ const Row = memo(
+              setActive(index);
             }
-            border={cellProps.pinned.border || cellProps.border}
-            pinnedOffset={pinnedOffset?._grommetDataTableSelect}
-            aria-disabled={isDisabled || !onSelect || undefined}
-            column={{
-              pin: Boolean(pinnedOffset?._grommetDataTableSelect),
-              plain: 'noPad',
-              size: 'auto',
-              render: () => (
-                <CheckBox
-                  tabIndex={onClickRow === 'select' ? -1 : undefined}
-                  a11yTitle={`${
-                    isSelected ? 'unselect' : 'select'
-                  } ${primaryValue}`}
-                  checked={isSelected}
-                  disabled={isDisabled || !onSelect}
-                  onChange={() => {
-                    if (isSelected) {
-                      onSelect(
-                        selected.filter((s) => s !== primaryValue),
-                        datum,
-                      );
-                    } else onSelect([...selected, primaryValue], datum);
-                  }}
-                  pad={cellProps.pad}
-                />
-              ),
-            }}
-            verticalAlign={verticalAlign}
-          />
-        )}
-        {rowDetails && (
-          <ExpanderCell
-            background={isSelected && cellProps.selected.background}
-            context={isRowExpanded ? 'groupHeader' : 'body'}
-            expanded={isRowExpanded}
-            expandLabel={expandLabel}
-            onToggle={() => {
-              let nextRowExpand;
-              const rowKey = primaryValue || index;
-              if (isRowExpanded) {
-                nextRowExpand = rowExpand.filter((s) => s !== rowKey);
-              } else {
-                nextRowExpand = [...rowExpand, rowKey];
+          }}
+          onMouseOver={
+            onClickRow && !isDisabled ? () => setActive(index) : undefined
+          }
+          tabIndex={
+            // eslint-disable-next-line no-nested-ternary
+            !onClickRow
+              ? undefined
+              : // If this row is focused, it should be focusable
+              // If no row is focused and this is the first row,
+              // make it focusable
+              (focused !== undefined && focused === index) ||
+                (focused === undefined && lastFocused === index) ||
+                (focused === undefined && index === 0)
+              ? 0
+              : // Otherwise, not in tab order
+                -1
+          }
+          {...passThemeFlag}
+        >
+          {(selected || onSelect) && (
+            <Cell
+              background={
+                (isSelected && cellProps.selected.background) ||
+                (active &&
+                  columns.pin &&
+                  theme.dataTable?.pinned?.body?.hover?.background) ||
+                (active && cellProps.active?.background) ||
+                (columns.pin && cellProps.pinned.background) ||
```

Comment:
Previously we were checking for `pinnedOffset` I'm just wondering why we don't check for that anymore